mailpoet/lib/WPCOM/DotcomLicenseProvisioner.php

```diff
@@ -0,0 +1,134 @@
+  ) {
+    $this->loggerFactory = $loggerFactory;
+    $this->settings = $settings;
+    $this->services = $services;
+  }
+
+  /**
+   * Returns true if in the context of WordPress.com Atomic platform.
+   *
+   * @return bool
+   */
+  public function isAtomicPlatform(): bool {
+    return defined('IS_ATOMIC') && IS_ATOMIC && defined('ATOMIC_CLIENT_ID') && (ATOMIC_CLIENT_ID === '2');
+  }
+
+  /**
+   * Activates MSS and adds API key for subscriptions purchased from WP.com Marketplace.
+   *
+   * @param bool $result
+   * @param array $licensePayload
+   * @param string $eventType
+   * @return bool|WP_Error
+   */
+  public function provisionLicense(bool $result, array $licensePayload, string $eventType) {
+    if (!$this->isAtomicPlatform() || $eventType !== self::EVENT_TYPE_PROVISION_LICENSE) {
+      return $result;
+    }
+
+    $apiKey = $this->getKeyFromPayload($licensePayload);
+    if (is_wp_error($apiKey)) {
+      $this->loggerFactory->getLogger(LoggerFactory::TOPIC_PROVISIONING)->error(
+        'key was not found in license payload');
+      return $apiKey;
+    }
+
+    return $this->activateMSS($apiKey);
+  }
+
+  /**
+   * Returns API key from license payload.
+   * @param array $licensePayload
+   * @return string|WP_Error
+   */
+  private function getKeyFromPayload(array $licensePayload) {
+    if (isset($licensePayload['apiKey']) && is_string($licensePayload['apiKey'])) {
+      return $licensePayload['apiKey'];
+    }
+
+    return new WP_Error('invalid_license_payload', 'Invalid license payload: Missing API key.');
+  }
+
+  /**
+   * Saves the API key and activates MSS.
+   * @param string $apiKey
+   * @return true|WP_Error
+   */
+  public function activateMSS(string $apiKey) {
+    $response = $this->settings->setupMSS($apiKey);
+    if ($response instanceof ErrorResponse) {
+      $this->loggerFactory->getLogger(LoggerFactory::TOPIC_PROVISIONING)->error(
+        'Setting sending method and key failed',
+        ['$response' => $response]
+      );
+      return new WP_Error('Provisioning failed setting the data', $this->concatMessages($response));
+    }
+
+    // This is necessary if the key changed but the sending method was already set to MailPoet
+    $response = $this->services->refreshMSSKeyStatus();
+    if ($response instanceof ErrorResponse) {
+      $this->loggerFactory->getLogger(LoggerFactory::TOPIC_PROVISIONING)->error(
+        'Refreshing the key failed',
+        ['$response' => $response]
+      );
+      return new WP_Error('Provisioning failed activating the data', $this->concatMessages($response));
+    }
+
+    $response = $this->services->refreshPremiumKeyStatus();
+    if ($response instanceof ErrorResponse) {
+      $this->loggerFactory->getLogger(LoggerFactory::TOPIC_PROVISIONING)->error(
+        'Refreshing Premium key failed',
+        ['$response' => $response]
+      );
+      return new WP_Error('Provisioning failed activating the data', $this->concatMessages($response));
```

Comment:
Hmm I've just realized that some plans that we offer in the shop has access to MSS but not to premium. Basically all Starter plans. Also Creator plans have only Premium and but not MSS.

I'm not sure if we offer those plans on WPCOM, but if we do we should make sure that the provisioning doesn't fail (end with WP_Error) for those. 

I see that we fixed this issue for the front end validation in https://github.com/mailpoet/mailpoet/pull/4399 and if I'm not mistaken the fix should also work for both `refreshPremiumKeyStatus` and `refreshMSSKeyStatus` because they internally reuse checkMSSKey respectively `checkPremiumKey` and if I read the code correctly they should retun Success response for `Bridge::KEY_VALID_UNDERPRIVILEGED`.

I rather keep this comment. I'm not sure if you are aware/tested the provisioning for such plan types.

Reply:
Thanks so much for taking the different cases into account!
We will have Starter and Business but not the other plans.
After your comment, I have also tested with the Starter plan, with and without the premium plugin.
The fix works as you said.

Reply:
Thanks for testing the started plan!